apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -383,6 +383,11 @@ type DatacenterOptions struct {
 
 	// +optional
 	DseWorkloads *cassdcapi.DseWorkloads `json:"dseWorkloads,omitempty"`
+
+	// Kubernetes resource requests and limits for the server-config-init init-container that is created by cass-operator to write the Cassandra config files.
+	// The default memory limit of 256Mi is sufficient for most use cases, but there are reports of OOM Kills which require to make this configurable.
+	// +optional
+	ConfigBuilderResources *corev1.ResourceRequirements `json:"configBuilderResources,omitempty"`
```

Comment:
Instead of exposing another field in an already lengthy spec, shouldn't we instead detect that a provided init container has the `"server-config-init"` name, and if so, extract any resource requirements it may contain, and put them in the right place in the CassandraDatacenter spec?

Alternatively: I also wonder if this isn't a bug in cass-operator. In general it merges any provided containers with the default ones, but here indeed it's doing something different and ignoring the resource requirements completely.

Reply:
> Instead of exposing another field in an already lengthy spec, shouldn't we instead detect that a provided init container has the "server-config-init" name, and if so, extract any resource requirements it may contain, and put them in the right place in the CassandraDatacenter spec?

Very good suggestion, let me try that